jormungandr/src/notifier/mod.rs

```diff
@@ -0,0 +1,180 @@
+use crate::intercom::NotifierMsg as Message;
+use crate::utils::async_msg::{channel, MessageBox, MessageQueue};
+use crate::utils::task::TokioServiceInfo;
+use chain_impl_mockchain::header::HeaderId;
+use futures::{SinkExt, StreamExt};
+use serde::{Serialize, Serializer};
+use slog::Logger;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+const MAX_CONNECTIONS_DEFAULT: usize = 255;
+
+// error codes in 4000-4999 are reserved for private use.
+// I couldn't find an error code for max connections, so I'll use the first one for now
+// maybe using the standard error code for Again is the right thing to do
+const MAX_CONNECTIONS_ERROR_CLOSE_CODE: u16 = 4000;
+const MAX_CONNECTIONS_ERROR_REASON: &str = "MAX CONNECTIONS reached";
+
+#[derive(Clone)]
+pub struct Notifier {
+    next_user_id: Arc<AtomicUsize>,
+    clients: Arc<tokio::sync::Mutex<Clients>>,
+    max_connections: usize,
+}
+
+#[derive(Serialize)]
+enum JsonMessage {
+    #[serde(serialize_with = "to_hex")]
+    NewBlock(HeaderId),
+    #[serde(serialize_with = "to_hex")]
+    NewTip(HeaderId),
+}
```

Comment:
This would live in better in jormungandr-lib

Reply:
moved